griptape/drivers/vector/peristable_local_vector_store_driver.py

```diff
@@ -0,0 +1,83 @@
+import dataclasses
+import json
+from pathlib import Path
+from typing import Union, Literal, cast, Dict
+from zipfile import ZipFile
+
+from attr import define, field
+
+from griptape.drivers import LocalVectorStoreDriver
+
+
+class EntryAwareJSONEncoder(json.JSONEncoder):
+    def default(self, o):
+        if isinstance(o, PersistableLocalVectorStoreDriver.Entry):
+            d_dict = dataclasses.asdict(o)
+            d_dict["__Entry__"] = True
+            return d_dict
+        return super().default(o)
+
+
+def as_entry(dct: Dict):
+
+    if "__Entry__" in dct:
+        dct.pop("__Entry__")
+        return PersistableLocalVectorStoreDriver.Entry(**dct)
+
+    return dct
```

Comment:
Could you move those inside of `PersistableLocalVectorStoreDriver`? We try to avoid using functions as well as multiple classes per file.

Reply:
Sure if that is your style. I think the most idiomatic way would be to make them underscore methods / classes and not put them inside `PersistableLocalVectorStoreDriver ` , but in the end it does not matter. 

I am on vacation the next week, so change will arrive 2nd week if august.